src/vis/bar/utils.ts

```diff
@@ -0,0 +1,240 @@
+import {merge} from 'lodash';
+import {CategoricalColumn, ColumnInfo, EColumnTypes, ESupportedPlotlyVis, IVisConfig, NumericalColumn, Scales} from '../interfaces';
+import {PlotlyInfo, PlotlyData} from '../interfaces';
+import {getCol} from '../sidebar/utils';
+
+export enum EBarDisplayType {
+    DEFAULT = 'Default',
+    NORMALIZED = 'Normalized',
+}
+
+export enum EBarDirection {
+    VERTICAL = 'Vertical',
+    HORIZONTAL = 'Horizontal',
+}
+
+export enum EViolinOverlay {
+    NONE = 'None',
+    STRIP = 'Strip',
+    BOX = 'Box'
+}
+
+export enum EBarGroupingType {
+    STACK = 'Stacked',
+    GROUP = 'Grouped',
+}
+
+export function isBar(s: IVisConfig): s is IBarConfig {
+    return s.type === ESupportedPlotlyVis.BAR;
+}
+
+export interface IBarConfig {
+    type: ESupportedPlotlyVis.BAR;
+    multiples: ColumnInfo | null;
+    group: ColumnInfo | null;
+    direction: EBarDirection;
+    display: EBarDisplayType;
+    groupType: EBarGroupingType;
+    numColumnsSelected: ColumnInfo[];
+    catColumnsSelected: ColumnInfo[];
+}
+
+const defaultConfig: IBarConfig = {
+    type: ESupportedPlotlyVis.BAR,
+    numColumnsSelected: [],
+    catColumnsSelected: [],
+    group: null,
+    groupType: EBarGroupingType.STACK,
+    multiples: null,
+    display: EBarDisplayType.DEFAULT,
+    direction: EBarDirection.VERTICAL,
+};
+
+export function barMergeDefaultConfig(
+    columns: (NumericalColumn | CategoricalColumn)[],
+    config: IBarConfig
+): IVisConfig {
+    const merged = merge({}, defaultConfig, config);
+
+    const catCols = columns.filter((c) => c.type === EColumnTypes.CATEGORICAL);
+
+    if(merged.catColumnsSelected.length === 0 && catCols.length > 0) {
+        merged.catColumnsSelected.push(catCols[catCols.length - 1].info);
+    }
+
+    return merged;
+}
+
+export function createBarTraces(
+    columns: (NumericalColumn | CategoricalColumn)[],
+    config: IBarConfig,
+    scales: Scales,
+): PlotlyInfo {
+    let counter = 1;
+
+    if(!config.catColumnsSelected) {
+        return {
+            plots: [],
+            legendPlots: [],
+            rows: 0,
+            cols: 0,
+            errorMessage: 'To create a Bar Chart, please select at least 1 categorical column.',
+            formList: ['groupBy', 'barMultiplesBy', 'barDirection', 'barGroupType', 'barNormalized']
+        };
```

Comment:
```suggestion
		throw new Error('To create a Bar Chart, please select at least 1 categorical column.');
```

In case of error it would make sense to throw errors and handle them in a parent component. It might be also useful to create a custom `VisError` class (somewhere) that can provide custom properties if necessary:

```ts
export class VisError extends Error {

}
```

If you wanna keep the object like you have it: do we still need the `fromList` property?

Please address for the other visualizations accordingly.

Reply:
I agree with this, though do worry slightly about making parent components deal with the errors. But I think as long as there is a sensible default for what happens that can be overridden via props, this is a great idea. But it will be left to a future PR